crates/katana/storage/provider/src/providers/fork/backend.rs

```diff
@@ -242,6 +268,33 @@ where
             }
         }
     }
+
+    fn dedup_request<F>(
+        &mut self,
+        req_key: BackendRequestIdentifier,
+        sender: OneshotSender<BackendResponse>,
+        rpc_call_future: F,
+    ) where
+        F: FnOnce() -> BoxFuture<'static, BackendResponse>,
+    {
+        if let std::collections::hash_map::Entry::Vacant(e) = self.request_dedup_map.entry(req_key)
```

Comment:
there shouldn't be a conflict so prefer to do it this way

```suggestion
        if let Entry::Vacant(e) = self.request_dedup_map.entry(req_key)
```